Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDll.cs

```diff
@@ -1243,6 +1260,312 @@ public DebugEventInfo GetLastEventInfo()
             }
         }
 
+        /// <summary>
+        /// Releases the process that is being debugged.
+        /// </summary>
+        public void ContinueExecution(Process process)
+        {
+            using (var processSwitcher = new ProcessSwitcher(StateCache, process))
+            {
+                DebugeeFlowControler flowControler = debugeeFlowControlers[process.Id];
+                flowControler.DebugStatusBreak.WaitOne();
+                Control.Execute(0, "g", 0);
+            }
+        }
+
+        /// <summary>
+        /// Continues execution of current process.
+        /// </summary>
+        public void ContinueExecution()
+        {
+            ContinueExecution(Process.Current);
+        }
+
+        /// <summary>
+        /// Breaks the process that is being debugged.
+        /// </summary>
+        public void BreakExecution(Process process)
+        {
+            using (var processSwitcher = new ProcessSwitcher(StateCache, process))
+            {
+                DebugeeFlowControler flowControler = debugeeFlowControlers[process.Id];
+                flowControler.DebugStatusBreak.Reset();
+                Control.SetInterrupt(0);
+                flowControler.DebugStatusBreak.WaitOne();
+
+                // Drop the cache.
+                //
+                process.InvalidateProcessCache();
+            }
+        }
+
+        /// <summary>
+        /// Break execution of current process.
+        /// </summary>
+        public void BreakExecution()
+        {
+            BreakExecution(Process.Current);
+        }
+
+        /// <summary>
+        /// Terminates the process that is being debugged and ends the session.
+        /// </summary>
+        public void Terminate(Process process)
+        {
+            Client.EndSession((uint)Defines.DebugEndActiveTerminate);
+
+            DebugeeFlowControler flowControler;
+            debugeeFlowControlers.RemoveEntry(process.Id, out flowControler);
+
+            flowControler.DebugStatusBreak.WaitOne();
+
+            // Release any threads that are waiting.
+            //
+            flowControler.DebugStatusGo.Set();
+            flowControler.DebugStatusBreak.Set();
+
+            flowControler.WaitForDebuggerLoopToExit();
+        }
+
+        /// <summary>
+        /// Terminates current process.
+        /// </summary>
+        public void Terminate()
+        {
+            Terminate(Process.Current);
+        }
+
+        #region Interactive debugger
+
+        /// <summary>
+        /// Controler for Debugee actions during live debugging.
+        /// </summary>
+        class DebugeeFlowControler
+        {
+            /// <summary>
+            /// Signal fired during interactive process debugging when debugee is released.
+            /// </summary>
+            public System.Threading.AutoResetEvent DebugStatusGo
+            {
+                get; private set;
+            }
```

Comment:
Done.

---
In reply to: [69631671](https://github.com/southpolenator/WinDbgCs/pull/8#discussion_r69631671) [](ancestors = 69631671)